src/UBAFeeCalculator/UBAFeeConfig.ts

```diff
@@ -76,6 +79,30 @@ class UBAConfig {
     this.lpGammaFunction = lpGammaFunction;
     this.incentivePoolAdjustment = incentivePoolAdjustment;
     this.ubaRewardMultiplier = ubaRewardMultiplier;
+
+    // Validate the config
+    this.assertValidityOfAllFeeCurves();
+  }
+
+  /**
+   * Assert the validity of all fee curves. This is a helper function
+   * that is called in the constructor to ensure that all fee curves
+   * are valid.
+   */
+  private assertValidityOfAllFeeCurves(): void {
+    // Find all the fee curves that could possibiliy be used
+    // in the UBA fee calculation. Specifically, these are the
+    // balancing fee curve and the lp gamma function curve. The
+    // curves are available for all overrides as well as their
+    // default counterparts.
+    const allBalancingFlows = [
+      this.balancingFee.default,
+      ...Object.values(this.balancingFee.override ?? {}),
+      this.lpGammaFunction.default,
```

Comment:
Does the lpGammaFunction need to have the same requirements as the balancing fee? Specifically, does it need to meet the requirement that it passes the 0 y-value at some point?

Reply:
I will check with Chase